Sources/StreamChat/Controllers/ChannelListController/ChannelListController.swift

```diff
@@ -190,178 +244,81 @@ public class ChatChannelListController: DataController, DelegateCallable, DataSt
-                log.debug("Channel list query has not yet created \(self.query)")
-                return
-            }
-
-            for channel in channels {
-                guard let channelDTO = session.channel(cid: channel.cid) else {
-                    log.error("Channel \(channel.cid) cannot be found in database.")
-                    continue
-                }
-                queryDTO.channels.insert(channelDTO)
-            }
-        } completion: { [weak self] _ in
-            let cids = channels.map(\.cid)
-            self?.worker.startWatchingChannels(withIds: cids) {
-                guard let error = $0 else { return }
-
-                log.warning(
-                    "Failed to start watching linked channels: \(cids), error: \(error.localizedDescription)"
-                )
-            }
+    /// Checks if the given channel should belong to the current query or not.
+    private func shouldChannelBelongToCurrentQuery(_ channel: ChatChannel) -> Bool {
+        if let filter = filter {
+            return filter(channel)
         }
-    }
-
-    private func unlink(channels: [ChatChannel]) {
-        guard !channels.isEmpty else { return }
-        client.databaseContainer.write { session in
-            guard let queryDTO = session.channelListQuery(filterHash: self.query.filter.filterHash) else {
-                log.debug("Channel list query has not yet created \(self.query)")
-                return
-            }
 
-            for channel in channels {
-                guard let channelDTO = session.channel(cid: channel.cid) else {
-                    log.error("Channel \(channel.cid) cannot be found in database.")
-                    continue
-                }
-                queryDTO.channels.remove(channelDTO)
-            }
+        guard !client.config.isChannelAutomaticFilteringEnabled else {
+            // When auto-filtering is enabled the channel will appear or not automatically if the
+            // query matches the DB Predicate. So here we default to saying it always belong to the current query.
+            return true
         }
-    }
-
-    // MARK: - Actions
 
-    /// Loads next channels from backend.
-    ///
-    /// - Parameters:
-    ///   - limit: Limit for page size.
-    ///   - completion: The completion. Will be called on a **callbackQueue** when the network request is finished.
-    ///                 If request fails, the completion will be called with an error.
-    ///
-    public func loadNextChannels(
-        limit: Int? = nil,
-        completion: ((Error?) -> Void)? = nil
-    ) {
-        if hasLoadedAllPreviousChannels {
-            completion?(nil)
-            return
+        // Fallback to legacy of checking if a channel should be linked to the current channel list query.
+        let deprecatedFallback: () -> Bool? = {
+            self.delegate?.controller(self, shouldAddNewChannelToList: channel)
```

Comment:
Why did you move from the switch checking the type of change, to this that is not using `shouldListUpdatedChannel`

Reply:
The reason is that it is a bit redundant. The logic of the `shouldAddNewChannelToList` and the `shouldListUpdatedChannel` is the same. We just want to verify if a channel should belong to a query, and so these 2 methods are not needed, only 1. Either way, these methods are deprecated for quite a while, so customers should not be using these already. (Hopefully)